### PR TITLE
replace http by https where deemed suitable

### PR DESCRIPTION
**Pull request by nbehrnd**
## Description
Separate from this git monitored repository, `markdownlint`
entered Linux Debian as package `ruby-mdl`.  During this work,
some files relevant to the packaging used outdated http addresses
now replaced by the https protocol now in place.
    
To preserve the functionality of markdownlint, there still are
seven untouched markdown files with examples and addresses of
the elder format 'http://'.  These include file `/docs/RULES.md`.

## Types of changes

- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/markdownlint/markdownlint/blob/master/CONTRIBUTING.md) document.
- [x] Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
- [x] Feature branch is up-to-date with `master`, if not - rebase it
- [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jaymzh) [inline CONTRIBUTING.md]: please revert this. Markdown best practices is to use all `1.` so that when you add/reorder, it all just renders properly.

further, this isn't in the description of your PR

File: CONTRIBUTING.md
Comment (jaymzh) [inline CONTRIBUTING.md]: revert

**Review by jaymzh — CHANGES_REQUESTED** [review]
The `s/http/https/g` changes are great, please revert the rest.

**Review by nbehrnd — COMMENTED** [review]
Your are right; I forgot that the assistance for sorted lists markdown offers.  Changes to the lists are reverted to the previous state,
and I approve the change.

File: CONTRIBUTING.md
Comment (nbehrnd) [reply CONTRIBUTING.md]: Addressed by commit [536c772](https://github.com/markdownlint/markdownlint/pull/445/commits/536c7721a80ccc0219ab2725b306539cf020cf37)

**Comment by nbehrnd** [comment]
@jaymzh 

For now 2023-02-04 Sat, GitHub's web interface reads 

> 2 workflows awaiting approval

The correction of the initial edit was filed, including a comment stating my approval in the message.  It however is not possible *for me* to click on the approve button, because I was the original author:

barrier

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Approved workflows (once you've got one PR merged, those will happen automatically), and approved the code changes.

You are missing DCO, so you'll need that.

**Comment by nbehrnd** [comment]
I forgot this detail and file a new PR https://github.com/markdownlint/markdownlint/pull/447 instead of this one.
And because the sign-off is an agreement among the contributors of this project, there will be an other one, too.

**Comment by jaymzh** [comment]
In future, please do not make a new PR, it loses all the history, when someone finds the PR that was merged. Instead just amend your commit and force-push.

**Pull request closed without merging**
